core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -24,6 +24,15 @@
 #include <cstdint>
 #include <string>
 
+// Use unlikely attribute from C++20 to improve performance of
+// reference counting when copy constructors are involved.
+#if defined(KOKKOS_ENABLE_IMPL_REF_COUNT_BRANCH_UNLIKELY) && \
+    !defined(KOKKOS_ENABLE_CXX17)
+#define KOKKOS_BRANCH_PROB [[unlikely]]
```

Comment:
Is it intentional to make it a public macro?
Also why give it that somewhat cryptic name rather than KOKKOS_IMPL_UNLIKELY_ATTRIBUTE

Reply:
If we don't undef them, the right place to define them would be Kokkos_Macros.hpp

Reply:
I put the macro in Kokkos_Macros.hpp and made other changes too.